app/models/component/manifest/app_container.rb

```diff
@@ -37,6 +37,7 @@ def environment_settings
         env << { name: '_JAVA_OPTIONS', value: java_options.to_s }.with_indifferent_access
         env << { name: '_BUSBAR_BUILD_TIME', value: timestamp.to_s }.with_indifferent_access
         env << { name: 'PORT', value: app_port.to_s }.with_indifferent_access
+        env << { name: 'DD_AGENT_HOST', valueFrom: fieldRef: fieldPath: 'status.hostIP'}.with_indifferent_access
```

Comment:
Should this be namespaced like the rest of the Busbar Server environment variables? This way it won't be overwritten by an applications' environment variables configuration.

Do you mind not abbreviating Datadog as `DD`? Unexpected abbreviations lead to confusion. So while `APP` and `ENV` are pretty obivous from context, `DD` would require finding this commit and either looking at its title or cross-referencing it with the corresponding `CHANGELOG` entry.

Since "host" could mean IP or (host) name, perhaps it's better to be more specific with `..._HOST_IP`? That's from the fact that the value is set to `status.hostIP` specifically.

Therefore, my recommendation is:
```suggestion
        env << { name: 'BUSBAR_DATADOG_AGENT_HOST_IP', valueFrom: fieldRef: fieldPath: 'status.hostIP'}.with_indifferent_access
```

Reply:
No because datadog APM agent expect the variable to have this specifc name. =(

Reply:
I see it:

> Your application tracers must be configured to submit traces to this address. Set environment variables with the `DD_AGENT_HOST` as the Agent container name, and `DD_TRACE_AGENT_PORT` as the Agent Trace port in your application containers. (In this example case, `datadog-agent` and `8126`, respectively.)

(Source: https://docs.datadoghq.com/agent/docker/apm/)

That's unfortunate, but it looks like there's not much we can do.

Does it mean you also need to set the `DD_TRACE_AGENT_PORT` then?

Reply:
AFAIK the dd lib will set this by default if not explicitly set! =P